### PR TITLE
Work anghul drone utils and instance

### PR DESCRIPTION
**Pull request by Anghul**
Creation of classes to allow the connection of the application with a simulation, and some utility functions for the drone

### CODE REVIEW

**Comment by jonasblanc** [comment]
A description would be useful to quickly understand the PR

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/drone/DroneInstanceMock.kt [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import io.mavsdk.mission.Mission
+import io.mavsdk.telemetry.Telemetry
+import io.reactivex.Completable
+import io.reactivex.Flowable
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito
+import org.mockito.Mockito.`when`
+
+object DroneInstanceMock {
+    val droneSystem: System = Mockito.mock(System::class.java)
```
Comment (jonasblanc) [inline app/src/androidTest/java/ch/epfl/sdp/drone3d/drone/DroneInstanceMock.kt:15]: We used the convention to import Mockito.* and not use Mockito. ... in code

File: app/src/main/java/ch/epfl/sdp/drone3d/MainActivity.kt [hunk 2]
```diff
@@ -19,9 +22,37 @@ class MainActivity : AppCompatActivity() {
     @Inject
     lateinit var authService: AuthenticationService
 
+    @SuppressLint("SetTextI18n")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main_menu)
+        if(isConnected()) {
+            val connectDroneButtonText = findViewById<Button>(R.id.connect_drone_button).apply {
+                text = "See connected drone"
```
Comment (jonasblanc) [inline app/src/main/java/ch/epfl/sdp/drone3d/MainActivity.kt:31]: The button text should not be in the code

File: app/src/main/res/values-round/strings.xml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<resources>
```
Comment (jonasblanc) [inline app/src/main/res/values-round/strings.xml:1]: Why is this file needed ?

File: app/src/main/res/values/dimens.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (jonasblanc) [inline app/src/main/res/values/dimens.xml:1]: Why is this file needed ?

**Review by jonasblanc — CHANGES_REQUESTED** [review]
We should probably state which files we took from Fly2Find

File: app/build.gradle [hunk 5]
```diff
@@ -51,6 +51,8 @@ dependencies {
     implementation 'com.google.code.gson:gson:2.8.6'
     // Hilt
     implementation "com.google.dagger:hilt-android:$hilt_version"
+    implementation 'androidx.wear:wear:1.1.0'
+    compileOnly 'com.google.android.wearable:wearable:2.8.1'
```
Comment (GabrielFleischer) [inline app/build.gradle:55]: I don't think those libraries are needed

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -18,5 +19,16 @@
...
+    <string name="drone_connect_port_default">Port</string>
+    <string name="drone_disconnect_button">Disconnect</string>
+    <string name="drone_simulated_ip">IP : 0.0.0.0</string>
+    <string name="drone_simulated_port">Port : 0</string>
     <string name="mapbox_access_token">pk.eyJ1IjoiZDNkIiwiYSI6ImNrbTRrc244djA1bGkydXRwbGphajZkbHAifQ.T_Ygz9WvhOHjPiOpZEJ8Zw</string>
+    <!--
+    This string is used for square devices and overridden by hello_world in
+    values-round/strings.xml for round devices.
+    -->
+    <string name="hello_world">Hello Square World!</string>
```
Comment (GabrielFleischer) [inline app/src/main/res/values/strings.xml:33]: This can be removed

**Review by GabrielFleischer — CHANGES_REQUESTED** [review]

File: app/src/main/res/values-round/strings.xml (hunk 3 above)
Comment (Anghul) [reply app/src/main/res/values-round/strings.xml:1]: Nothing, it was created by error

File: app/src/main/res/values/dimens.xml (hunk 4 above)
Comment (Anghul) [reply app/src/main/res/values/dimens.xml:1]: Nothing, it was created by error

**Review by GabrielFleischer — APPROVED** [review]

**Review by GabrielFleischer — APPROVED** [review]

**Review by jonasblanc — APPROVED** [review]

**Pull request merged**